src/dataviews/category-dataview-model.js

```diff
@@ -0,0 +1,339 @@
+var _ = require('underscore');
+var DataviewModelBase = require('./dataview-model-base');
+var SearchModel = require('./category-dataview/search-model');
+var CategoryModelRange = require('./category-dataview/category-model-range');
+var CategoriesCollection = require('./category-dataview/categories-collection');
+var LockedCatsCollection = require('./category-dataview/locked-categories-collection');
```

Comment:
IMO, some of the logic in ^^ models shouldn't be part of the dataview. For example, category widgets should use a category-dataview-model to "fetch" the search results, but the dataview itself shouldn't know much about search results. Or the dataview shouldn't know what categories are locked, etc.